influxql/select.go

```diff
@@ -311,8 +311,9 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions) (Iter
 			interval := opt.DerivativeInterval()
 			isNonNegative := (expr.Name == "non_negative_derivative")
 
-			// Derivatives do not use GROUP BY intervals, so clear this option.
+			// Derivatives do not use GROUP BY intervals or time constraints, so clear these options.
```

Comment:
Is this because the group by and time intervals have already been setup and derivative should just operate over what is passed to it?